libraries/AP_Networking/AP_Networking_port.cpp

```diff
@@ -368,22 +384,22 @@ bool AP_Networking::Port::send_receive(void)
             WITH_SEMAPHORE(sem);
             n = writebuffer->peekbytes(buf, available);
         }
-        if (n > 0) {
-            const auto ret = sock->send(buf, n);
+        if (n > 0 &&
```

Comment:
just move the ret declaration before the if() so the advance() code can be in common and not duplicated

Reply:
fixed